src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.wasm.cs

```diff
@@ -72,5 +74,29 @@ private void ChangeViewForOrientation(Orientation orientation, double scrollAdju
 				ChangeView(HorizontalOffset + scrollAdjustment, null, null, disableAnimation: true);
 			}
 		}
+
+		private protected override void OnLoaded()
+		{
+			base.OnLoaded();
+			AddHandler(KeyDownEvent, new KeyEventHandler(OnKeyDown), true);
+			AddHandler(PointerWheelChangedEvent, new PointerEventHandler(OnPointerWheelChanged), true);
+		}
+
+		private void OnPointerWheelChanged(object sender, PointerRoutedEventArgs e)
+		{
+			// There is no way to distinguish the cause of a `scroll` DOM event, so we need to discriminate between
+			// wheel scrolling and keyboard scrolling before we get the `scroll` event. We only need to cancel keyboard
+			// scrolling (due to space, etc.)
+			CancelNextNativeScroll = false;
+		}
+
+		private protected override void OnUnloaded()
+		{
+			base.OnUnloaded();
+			RemoveHandler(KeyDownEvent, new KeyEventHandler(OnKeyDown));
+			RemoveHandler(PointerWheelChangedEvent, new KeyEventHandler(OnKeyDown));
+		}
+
+		private void OnKeyDown(object sender, KeyRoutedEventArgs args) => CancelNextNativeScroll = args.Handled;
```

Comment:
I'm wondering why this is in the `ScrollViewer` instead of the `ScrollContentPresenter`? I have the feeling that we could keep all scroll handling in the presenter.

Reply:
Initially, that was indeed how I implemented it, but AFAIK, our behaviour on the other platforms is that if args are Handled at the ScrollViewer level, we still don't scroll, so we need to wait until we go to the ScrollViewer to see if we handle it or not.

Reply:
But we could move those event handlers in the SCP so all the logic will remain in the same class/file ?

Reply:
I guess we could subscribe to the ScrollViewer events from the SCP, but it would look awkward. It makes more sense to me for the ScrollViewer to handle its own `CancelNextNativeScroll` state and for the SCP to just read from it, but if you think moving everything to SCP makes more sense, we can go with that too.